### ISSUE TITLE
Please correct the SerializerInfo

### ISSUE THREAD

**Issue by itadapter** [kevin-montrose/Jil#141]
Kevin, we have included JIL in SERBENCH suite:
https://github.com/aumcode/serbench/blob/master/Source/Serbench.Specimens/Serializers/JilSerializer.cs

How many lines of code you have *ballpark*, and how many data types (i.e.classes, interfaces etc..)
we are trying to compare the complexity of various serializers,
Could you *please correct the fields below*(if any are wrong):
   
 [SerializerInfo( 
     Family = SerializerFamily.Textual,    
     MetadataRequirement = MetadataRequirement.None,
     VendorName = "Kevin Montrose",
     VendorLicense = "The MIT License (MIT)",
     VendorURL = "https://github.com/kevin-montrose/Jil",
     VendorPackageAddress = "Install-Package Jil",
     FormatName = "JSON",
     LinesOfCodeK = 0,                     
     DataTypes = 0,
     Assemblies = 1,
     ExternalReferences = 0,
     PackageSizeKb = 393
  )]

**Comment by itadapter** [comment]
sorry that I have reported this as an "issue", I could not find your email.
If you could just respond with KinesOfCode (average number) and DataTypes Count
that would be great. I could count that myself byt I dont know your code well enough to provide accurate numbers.  Also, does JIL use refs? I believe it uses SIGIL?

**Comment by Wazner** [comment]
Line counts include comments and empty lines (all lines in the *.cs files).
The number of types is retrieved using `Assembly.GetTypes().Length`, so it includes nested types.

**Number of files:** 117
**Total line count:** 63425
**Average line count:** ±542
**Number of types:** 323

And yes, Jil uses Sigil for generating executable code.

*IMHO: Line count and number of types are a bad indicator of program complexity.*

**Comment by itadapter** [comment]
Thanks for the numbers, I'll put them in SERBENCH inventory.
Line count is  a good relative ballpark indicator i.e. JIL: 60K (i'll round down) 300 types
NFX.JSON:  2K, 10 types
NFX Json is so small because it does not have to implement many things that are already implemented in NFX. Interestingly both pass the same tests although JIL is certainly faster, but that is not the purpose of NFX.JSON (it is used for serialization of TypedRows, not for arbitrary types), so it is a good relative comparison

**Comment by itadapter** [comment]
CLOSED as resolved

**Comment by itadapter** [comment]
btw,
 take a look at how JIL performs against others:
http://aumcode.github.io/serbench/
(click on TypicalPerson and ObjectGraph charts and tables) links

**Comment by Wazner** [comment]
After having a look at your benchmarks, I noticed you use Jil's SerializeDynamic and DeserializeDynamic (passing an object to Serialize also uses Dynamic serialization). These methods perform slower than the generic versions Serialize and Deserialize. Just something to keep in mind.

**Comment by itadapter** [comment]
The whole point of the test is in permuting different serializer facades with different payloads.
The payload is the key. Of course many other serializers would have performed way faster had we only serialized say TypicalPerson without polymorphism, i.e. our NFX.Slim serializer is slower than Protobuf only because of 100% dynamic type system, that is we can serialize object[] or Car[] filled with car descendants. 
Ad for JIL, I think it is a very solid piece of work and performs quite well in most cases, however I trully think that you guys CAN identify and optimize some things in JIL as our tests show.

For example, TypicalPerson[1] and TypicalPerson[50] shows disproportional performance drop in JIL, which leads me to believe that there is some hot spot, maybe pre-alloced buffer which gets resized to often after some threshold. Again, this is my rough thinking, I encourage you to do different payload combinations and see how performance get affected. 
In my case I have identified Array iteration as the weakest link via a lambda scope closure.
I have refactored 1D arrays into a linear loop and that gave me +10% perf in most cases (1d arrays)
